presto-main/src/test/java/com/facebook/presto/sql/planner/TestEqualityInference.java

```diff
@@ -318,6 +335,32 @@ public void testEqualityGeneration()
         assertEquals(scopedCanonical, nameReference("a1"));
     }
 
+    @Test
+    public void testExpressionsThatMayReturnNullOnNonNullInput()
+            throws Exception
+    {
+        List<Expression> candidates = ImmutableList.of(
```

Comment:
An assert helper wouldn't work out very well. The names used in the expressions are tightly coupled with the names used in the test code below.